selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/NewWorkspacePageTest.java

```diff
@@ -0,0 +1,519 @@
+    assertEquals(newWorkspace.getAvailableStacksCount(), EXPECTED_SINGLE_MACHINE_STACKS_COUNT);
+
+    newWorkspace.clickOnAllButton();
+    newWorkspace.waitStacks(EXPECTED_ALL_STACKS);
+    assertEquals(newWorkspace.getAvailableStacksCount(), EXPECTED_ALL_STACKS_COUNT);
+
+    newWorkspace.clickOnMultiMachineButton();
+    newWorkspace.waitStacks(EXPECTED_MULTI_MACHINE_STACKS);
+    assertEquals(newWorkspace.getAvailableStacksCount(), EXPECTED_MULTI_MACHINE_STACKS_COUNT);
+
+    newWorkspace.clickOnAllButton();
+    newWorkspace.waitStacks(EXPECTED_SINGLE_MACHINE_STACKS);
+    newWorkspace.waitStacks(EXPECTED_MULTI_MACHINE_STACKS);
+    assertEquals(
+        newWorkspace.getAvailableStacksCount(),
+        EXPECTED_SINGLE_MACHINE_STACKS_COUNT + EXPECTED_MULTI_MACHINE_STACKS_COUNT);
+
+    newWorkspace.clickOnQuickStartButton();
+    newWorkspace.waitStacksOrder(EXPECTED_QUICK_START_STACKS);
+    newWorkspace.clickNameButton();
+
+    try {
+      newWorkspace.waitStacksOrder(EXPECTED_QUICK_START_STACKS_REVERSE_ORDER);
+    } catch (TimeoutException ex) {
+      // remove try-catch block after issue has been resolved
+      fail("Known issue https://github.com/eclipse/che/issues/5650", ex);
+    }
+
+    newWorkspace.clickNameButton();
+    newWorkspace.waitStacksOrder(EXPECTED_QUICK_START_STACKS);
+  }
+
+  @Test
+  public void checkFiltersButton() {
+    newWorkspace.waitPageLoad();
+    newWorkspace.clickOnFiltersButton();
+    newWorkspace.waitFiltersFormOpened();
+    seleniumWebDriverHelper.sendKeys(ESCAPE.toString());
+    newWorkspace.waitFiltersFormClosed();
+
+    newWorkspace.clickOnFiltersButton();
+    newWorkspace.waitFiltersFormOpened();
+    newWorkspace.clickOnTitlePlaceCoordinate();
+    newWorkspace.waitFiltersFormClosed();
+
+    newWorkspace.clickOnFiltersButton();
+    newWorkspace.waitFiltersFormOpened();
+    newWorkspace.typeToFiltersInput("j");
+    newWorkspace.waitFiltersSuggestionsNames(EXPECTED_FILTERS_SUGGESTIONS);
+
+    assertEquals(
+        newWorkspace.getSelectedFiltersSuggestionName(),
+        newWorkspace.getFiltersSuggestionsNames().get(0));
+
+    seleniumWebDriverHelper.sendKeys(ARROW_DOWN.toString());
+    newWorkspace.waitSelectedFiltersSuggestion("JDK");
+
+    seleniumWebDriverHelper.sendKeys(ARROW_UP.toString());
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA");
+
+    seleniumWebDriverHelper.sendKeys(ARROW_UP.toString());
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA 1.8, THEIA");
+
+    newWorkspace.singleClickOnFiltersSuggestions("JAVA");
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA");
+
+    newWorkspace.singleClickOnFiltersSuggestions("JDK");
+    newWorkspace.waitSelectedFiltersSuggestion("JDK");
+
+    newWorkspace.doubleClickOnFiltersSuggestion("JDK");
+    newWorkspace.waitFiltersInputFieldTags(asList("JDK"));
+
+    newWorkspace.deleteLastTagFromInputTagsField();
+    newWorkspace.waitFiltersInputFieldIsEmpty();
+
+    newWorkspace.typeToFiltersInput("j");
```

Comment:
It's not quite clear why we are using "j" here, not "k" or "a". So, it would be clear to name `"j"` somehow to explain its propose.